src/fsharp/SyntaxTrivia.fs

```diff
@@ -4,6 +4,26 @@ namespace FSharp.Compiler.SyntaxTrivia
 
 open FSharp.Compiler.Text
 
+[<RequireQualifiedAccess; NoEquality; NoComparison>]
+type ConditionalDirectiveTrivia =
+    | IfDirectiveTrivia of expr:IfDirectiveExpression * range:range
+    | ElseDirectiveTrivia of range:range
+    | EndIfDirectiveTrivia of range:range
+
+and [<RequireQualifiedAccess; NoEquality; NoComparison>] IfDirectiveExpression =
+    | IfdefAnd of IfDirectiveExpression * IfDirectiveExpression
```

Comment:
Similarly this can just be

```
and [<RequireQualifiedAccess; NoEquality; NoComparison>] IfDirectiveExpression =
    | And of IfDirectiveExpression * IfDirectiveExpression
    | Or of IfDirectiveExpression * IfDirectiveExpression
    | Not of IfDirectiveExpression
    | Ident of string
```